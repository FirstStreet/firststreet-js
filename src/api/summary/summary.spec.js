const FirstStreet = require('../../__mocks__/FirstStreet');
const summaryPropertyMock = require('../../__mocks__/summaryProperty');
const summaryCityMock = require('../../__mocks__/summaryCity');

const validMethods = ['getPropertyByFSID', 'getCityByFSID', 'getPropertyByLatLng', 'getCityByLatLng', 'getPropertyByAddress', 'getCityByAddress'];

describe('dataSummary', () => {
  const fsf = new FirstStreet('aa.bb.cc', {
    host: 'https://FAKE-HOST-FOR-TESTING.com',
  ***REMOVED***);

  it('should contain valid methods', () => {
    validMethods.forEach((method) => {
      expect(fsf.dataSummary[method]).toBeDefined();
    ***REMOVED***);
  ***REMOVED***);

  it('.getPropertyByFSID should get property by ID', async () => {
    const property = await fsf.dataSummary.getPropertyByFSID(summaryPropertyMock.FSID);

    expect(property).toMatchSnapshot();
    expect(property.primaryNumber).toBe(summaryPropertyMock.results.location.primaryNumber);
  ***REMOVED***);

  it('.getCityById should get city by ID', async () => {
    const city = await fsf.dataSummary.getCityByFSID(summaryCityMock.ID);

    expect(city).toMatchSnapshot();
    expect(city.name).toBe(summaryCityMock.name);
  ***REMOVED***);

  it('.getPropertyByLatLng should get property by lat, lng', async () => {
    const property = await fsf.dataSummary.getPropertyByLatLng(summaryPropertyMock.results.location.geometry.center.coordinates[1], summaryPropertyMock.results.location.geometry.center.coordinates[0]);

    expect(property).toMatchSnapshot();
    expect(property.primaryNumber).toBe(summaryPropertyMock.results.location.primaryNumber);
  ***REMOVED***);

  it('.getCityByLatLng should get city by lat, lng', async () => {
    const city = await fsf.dataSummary.getCityByLatLng(summaryCityMock.results.location.geometry.center.coordinates[1], summaryCityMock.results.location.geometry.center.coordinates[0]);
    expect(city).toMatchSnapshot();
    expect(city.name).toBe(summaryCityMock.name);
  ***REMOVED***);

  it('.getPropertyByAddress should get property by address lookup', async () => {
    const property = await fsf.dataSummary.getPropertyByAddress('212 appoquin s, middletown, delware');

    expect(property).toMatchSnapshot();
    expect(property.primaryNumber).toBe(summaryPropertyMock.results.location.primaryNumber);
  ***REMOVED***);

  it('.getCityByAddress should get city by address lookup', async () => {
    const city = await fsf.dataSummary.getCityByAddress('middletown, delware');

    expect(city).toMatchSnapshot();
    expect(city.name).toBe(summaryCityMock.results.location.name);
  ***REMOVED***);
***REMOVED***);
