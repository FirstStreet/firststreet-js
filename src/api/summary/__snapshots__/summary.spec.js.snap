// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataSummary .getCityById should get city by ID 1`] = `
Object {
  "errors": true,
  "messages": "Expected required FSID. Usage: .getCityByFSID(fsid)",
***REMOVED***
`;

exports[`dataSummary .getCityByLatLng should get city by lat, lng 1`] = `
Object {
  "error": "data not found",
  "errors": true,
  "messages": null,
***REMOVED***
`;

exports[`dataSummary .getPropertyByAddress should get property by address lookup 1`] = `
Object {
  "error": "data not found",
  "errors": true,
  "messages": null,
***REMOVED***
`;

exports[`dataSummary .getPropertyByFSID should get property by ID 1`] = `
Object {
  "error": "data not found",
  "errors": true,
  "messages": null,
***REMOVED***
`;

exports[`dataSummary .getPropertyByLatLng should get property by lat, lng 1`] = `
Object {
  "error": "data not found",
  "errors": true,
  "messages": null,
***REMOVED***
`;
